src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -13,21 +13,180 @@
 #
 #
 
-.rs.addJsonRpcHandler("preview_data_import", function(path, maxRows, maxCols = 100, maxFactors = 64)
+.rs.addFunction("assemble_data_import_name", function(dataImportOptions)
+{
+  dataName <- dataImportOptions$dataName
+
+  if (is.null(dataName) || identical(dataName, ""))
+    {
+      if (!is.null(dataImportOptions$importLocation))
+      {
+        locationName <- basename(dataImportOptions$importLocation)
+        if (length(locationName) > 0)
+        {
+          fileComponents <- unlist(strsplit(locationName, ".", fixed = TRUE))
+          components <- length(fileComponents)
+          if (components >= 1)
+          {
+            dataName <- paste(fileComponents[1:components-1], collapse = "_")
```

Comment:
Double-checking precedence: should this be `1:(components - 1)`? Or do you really want to subtract 1 from the whole vector (ie, you want `(1:components) - 1`?)